rpcserver.go

```diff
@@ -6086,65 +6129,76 @@ func (s *rpcServer) jsonRPCRead(w http.ResponseWriter, r *http.Request, isAdmin
 	conn.SetReadDeadline(timeZeroVal)
 
 	// Attempt to parse the raw body into a JSON-RPC request.
-	var responseID interface{}
 	var jsonErr error
-	var result interface{}
-	var request dcrjson.Request
+	var msg []byte
+	var request interface{}
```

Comment:
To clarify a bit on what I mean about interfaces, notice the code here is doing the unmarshal into an interface.  Instead, it would be more efficient and safer to look at the byte stream in order to determine whether it should unmarshal into a `Request` or a `[]Request` in order to be properly typed.

For example:

```Go
// At package scope...
var batchedRequestPrefix = []byte("[]")
...

if bytes.HasPrefix(body, batchedRequestPrefix) {
	/* attempt unmarshal to []Request */
} else {
	/* attempt unmarshal to Request */
}
```